string method
bool idle
---
bool current_idle_status